server/src/client_api.rs

```diff
@@ -22,12 +25,15 @@ use unleash_types::client_metrics::{
 #[get("/client/features")]
 pub async fn features(
     edge_token: EdgeToken,
-    features_source: web::Data<dyn EdgeSource>,
+    features_cache: web::Data<DashMap<String, ClientFeatures>>,
 ) -> EdgeJsonResult<ClientFeatures> {
-    features_source
-        .get_client_features(&edge_token)
-        .await
+    features_cache
+        .get(&cache_key(edge_token))
+        .map(|features| features.clone())
         .map(Json)
+        .ok_or_else(|| {
+            EdgeError::PersistenceError("TODO: Don't use this error here!!!!!!!!".into())
```

Comment:
👀

Reply:
Yep, thank you!

We don't actually know what to do here right now. 200 with an empty response is the worst possible option because that will cause the calling SDK to purge it's cache. 500 isn't great either for obvious reasons. We'd like to pick a better error handling flow for this at some point but in the interest of keeping this PR a sane size we opted for this

I have, however, patched the error message to not look quite so insane